src/engine/source/hlp/src/SpecificParsers.cpp

```diff
@@ -376,3 +378,113 @@ bool parseURL(const char **it, char endToken, URLResult &result) {
 
     return true;
 }
+
+bool parseDomain(const char **it, char endToken, std::vector<std::string> const& captureOpts, DomainResult &result) {
+    static const std::string_view valid_chars {"abcdefghijklmnopqrstuvwxyz"
+                                               "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                                               "0123456789-_."};
+    static const int DOMAIN_MAX_SIZE = 253;
+    static const int LABEL_MAX_SIZE = 63;
+
+    const char *start = *it;
+    while (**it != endToken || **it != '\0') {
```

Comment:
this should be `while(((**it) != 0) && ((**it) != endToken))`

Reply:
You are right! It was a big mistake.
Good catch!